### DISCUSSION TITLE
Getting style error during commit unrelated to my change

### DISCUSSION THREAD

**Discussion by hasnainzeenwala** [pytest-dev/pytest#10037] [Q&A]
Hi!
I've made changes to _pytest/logging.py to resolve issue #7431. But while I'm committing my changes I'm getting the following error: 
```
src/_pytest/assertion/rewrite.py:284: error: Returning Any from function declared to return "TraversableResources"  [no-any-return]
Found 1 error in 1 file (checked 1 source file)

```
Not sure how this is related to my change or what I could do to fix this. 

Edit:  sorry for the misleading question. It is related to my change because I did a random commit on local master and didn't get this error, but I'm still unable to understand how my change in the logging.py file is related to this. I'm unable to trace it.

**Comment by portyu9** [comment]
Hi @hasnainzeenwala 

That `mypy` error is contributor/tooling related, not a runtime pytest failure, and the traceback alone is not enough to attribute it to the line you changed.

Since the same check passes on `master`, I would first determine whether the difference comes from the branch or from a stale pre-commit environment:

```bash
pre-commit clean
pre-commit run --all-files
```

Then try the same branch from a clean checkout/virtualenv using the repository's pinned development dependencies.

If it still fails, bisect or temporarily revert the commits on your branch until the error disappears. A type error in an apparently unrelated file can happen when another change modifies imports, dependency/type versions, or the inferred type graph.

So I would **not change `rewrite.py` merely to silence this error** unless the failure reproduces from a clean environment on the branch and the inferred return type there is actually wrong.

Please mark this answer as accepted if it helped, thank you.
